model/flow/epoch.go

```diff
@@ -397,22 +435,37 @@ func (commit *EpochCommit) EqualTo(other *EpochCommit) bool {
 	if commit.DKGGroupKey != nil && other.DKGGroupKey != nil && !commit.DKGGroupKey.Equals(other.DKGGroupKey) {
 		return false
 	}
+
 	if len(commit.DKGParticipantKeys) != len(other.DKGParticipantKeys) {
 		return false
 	}
-
 	for i, key := range commit.DKGParticipantKeys {
 		if !key.Equals(other.DKGParticipantKeys[i]) {
 			return false
 		}
 	}
 
+	if len(commit.DKGIndexMap) != len(other.DKGIndexMap) {
```

Comment:
We should be able to use https://pkg.go.dev/maps#Equal. It's available from Go 1.21. 
There is an equivalent one for slices too that can be useful in this same function